### PR TITLE
feat: add GPU banner to Podman AI Lab pages

### LINKED ISSUE #1591: Add banner about GPU acceleration if not used/configured
### Is your enhancement related to a problem? Please describe

People who are not aware of the possibility of using GPU acceleration will not notice the capability.

### Describe the solution you'd like

If the GPU is not configured in the user's environment, we'd like to drive more attention onto how the user could experiment with it.

- The banner should be displayed at the top of all screens of Podman AI Lab
- The title should be "⚡️ **Supercharge Your AI:**  Enable GPU Acceleration and watch your LLM respond in a flash! "
- There should be one button "Enable": when clicked, it displays the instructions on how the user can enable the GPU acceleration, depending on the platform the user is using.

It should be possible to hide the banner with a button: "Don't display anymore"

We should have telemetry on the banner to know:
- how often it's been displayed
- how often the user clicked on "enable" or "don't display anymore"

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeffmaury**
Fixes #1591

### What does this PR do?

Add a new banner to all Podman AI Lab pages 

### What issues does this PR fix or reference?

#1591

### How to test this PR?

1. Make sure the GPU experimental flag is off
2. Navigate to one of the Podman AI Lab pages and check the banner is displayed
3. Verify banner disappear if one of the button is clicked

### CODE REVIEW

**Review by axel7083 — COMMENTED** [review]
image

I must say this is very intrusive, and do not really fit well in the style .

The checkbox do not really make sense IMO, as when you click on it, the banner just disappear, checkbox means you can enabled/disabled something, here this is an action, and should be a button.

**Review by benoitf — COMMENTED** [review]
you might add a smaller PR having all the flags being introduced, new methods etc but not the UI

so at least we could merge early this part

File: packages/frontend/src/lib/notification/GPUPromotion.spec.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    },
+  };
+});
+
+const mocks = vi.hoisted(() => {
+  return {
+    getConfigurationMock: vi.fn<() => ExtensionConfiguration>(),
+  };
+});
+
```
Comment (benoitf) [inline packages/frontend/src/lib/notification/GPUPromotion.spec.ts:39]: I don't get why there the usage of vi.hoisted

AFAIK we don't need it

I commented about that on this issue: https://github.com/containers/podman-desktop-extension-ai-lab/pull/1849

File: packages/frontend/src/lib/notification/GPUPromotion.spec.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+vi.mock('../../stores/extensionConfiguration', () => ({
+  configuration: {
+    subscribe: (f: (msg: ExtensionConfiguration) => void) => {
+      f(mocks.getConfigurationMock());
+      return (): void => {};
+    },
+  },
+}));
```
Comment (benoitf) [inline packages/frontend/src/lib/notification/GPUPromotion.spec.ts:47]: you can mock more easily the store 
https://github.com/containers/podman-desktop-extension-ai-lab/pull/1849#discussion_r1795172109

File: packages/frontend/src/lib/notification/GPUPromotion.svelte [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+// eslint-disable-next-line quotes
+const actionName = "Don't display anymore";
+
+function hideGPUPromotionBanner(): void {
+  studioClient.updateExtensionConfiguration({ showGPUPromotion: false }).catch(console.error);
+}
+
+function enableGPUSupport(): void {
+  studioClient.updateExtensionConfiguration({ experimentalGPU: true }).catch(console.error);
+}
```
Comment (benoitf) [inline packages/frontend/src/lib/notification/GPUPromotion.svelte:16]: I don't see that being covered in the unit tests

**Review by benoitf — APPROVED** [review]
about the UI itself, it may evolve by follow-ups on my end

File: packages/frontend/src/lib/notification/GPUPromotion.svelte [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+const content =
+  'GPU support is not enabled. Podman AI Lab supports GPU on Windows on MacOS.\n\nThis greatly enhance the developer experience when running inference servers. We recommend you to enable it.';
```
Comment (Firewall) [inline packages/frontend/src/lib/notification/GPUPromotion.svelte:19]:
```suggestion
  'GPU support is not enabled. Podman AI Lab supports GPU on Windows on MacOS.\n\nThis greatly enhances the developer experience when running inference servers. We recommend you to enable it.';
```

**Review by Firewall — CHANGES_REQUESTED** [review]

**Review by Firewall — APPROVED** [review]

**Review by axel7083 — APPROVED** [review]

**Pull request merged**